### PR TITLE
Issue #13345: Enable examples tests for JavadocMethodCheck

### PR DESCRIPTION
**Pull request by AmitKumarDeoghoria**
### Title: Enable examples tests forJavadocMethodCheck
**Related Issue: #13345** 

**Summary**
This pull request aims to enable example tests for the **  JavadocMethodCheck** in the project. The implementation focuses on ensuring that   JavadocMethod checks are correctly validated through comprehensive test cases, improving overall code reliability and quality.

**Key Changes Made**

- Conversion of txt file to Java File
- Enable the test cases by adding expected violations in the expected array for all example tests
- Made corresponding changes in the xml and xml.template file
- Adding suppression for file length check for example7

### CODE REVIEW

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocmethod/Example1.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+*/
+package com.puppycrawl.tools.checkstyle.checks.javadoc.javadocmethod;
+
+// xdoc section -- start
+public class Example1 {
+
+  /**
+   *
+   */
+  Example1(int x) { // violation, param tag missing for x
```
Comment (romani) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocmethod/Example1.java:16]: Please do compact form of ctor 
`Example1(int x) {}`

Please use violation message in trailing comments, with single quotes.

Please make javadoc to be compact `/** */`
Please put violation comment under ctor, to be horizontally short.

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocmethod/Example1.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  }
+
+  /**
+   *
+   */
+  public int testMethod1(int p1) {  // violation, '@return tag should be present'
```
Comment (romani) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocmethod/Example1.java:22]: `public int m1(int p1) { return p1; }`
Violation comment under method.

Apply same updates to other.

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocmethod/Example1.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    return p1;  // violation 1 lines above 'Expected @param tag for 'p1''
+  }
+
+  /**
+   *
```
Comment (romani) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocmethod/Example1.java:27]: Remove this empty line

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocmethod/Example7.java [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+   */
+  public int parsePositiveInt(String s) {
+    try {
+      int value = Integer.parseInt(s);
+      if (value <= 0) {
+        throw new NumberFormatException(value + " is negative/zero"); // ok, try
+      }
+      return value;
+    } catch (NumberFormatException ex) {
+      // violation below, '@throws tag for 'IllegalArgumentException''
```
Comment (romani) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocmethod/Example7.java:62]: Please put "Expected" in message

**Review by romani — CHANGES_REQUESTED** [review]
Fix CI please.

Items:

**Review by romani — CHANGES_REQUESTED** [review]
Fix CI please.

Items:

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocmethod/Example7.java (hunk 4 above)
Comment (AmitKumarDeoghoria) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocmethod/Example7.java]: Done

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocmethod/Example1.java (hunk 3 above)
Comment (AmitKumarDeoghoria) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocmethod/Example1.java]: Done

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocmethod/Example1.java (hunk 2 above)
Comment (AmitKumarDeoghoria) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocmethod/Example1.java]: Done

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocmethod/Example1.java (hunk 1 above)
Comment (AmitKumarDeoghoria) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocmethod/Example1.java]: Done

**Comment by AmitKumarDeoghoria** [comment]
GitHub, generate website

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocmethod/Example1.java [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+public class Example1 {
+  /** */
+  Example1(int x) {}
+  // violation above, 'Expected @param tag for 'x''
+  /** */
+  public int testMethod1(int p1) {
+    // violation above, '@return tag should be present'
+    return p1; // violation 2 lines above 'Expected @param tag for 'p1''
+  }
+  /**
```
Comment (romani) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocmethod/Example1.java:20]: Please add empty line before javadoc

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocmethod/Example1.java [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+   * @param p1 The first number
+   */
+  @Deprecated
+  private int testMethod2(int p1) {
+    return p1; // violation 2 lines above '@return tag should be present'
+  }
+  /** */
```
Comment (romani) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocmethod/Example1.java:27]: lease add empty line before javadoc

**Review by romani — CHANGES_REQUESTED** [review]
Last:

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocmethod/Example1.java (hunk 6 above)
Comment (AmitKumarDeoghoria) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocmethod/Example1.java:30]: Done

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocmethod/Example1.java (hunk 5 above)
Comment (AmitKumarDeoghoria) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocmethod/Example1.java:22]: Done

**Comment by romani** [comment]
GitHub, generate website

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocmethod/Example1.java [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+public class Example1 {
+
+  /** */
+  Example1(int x) {}
+  // violation above, 'Expected @param tag for 'x''
+  /** */
+  public int testMethod1(int p1) {
+    // violation above, '@return tag should be present'
+    return p1; // violation 2 lines above 'Expected @param tag for 'p1''
+  }
```
Comment (romani) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocmethod/Example1.java:20]: please make body of all methods to be single line, their content does not matther
```
  /** */
  public int m1(int p1) { return p1; }
  // 2 violations above:
  //    '@return tag should be present'
  //    'Expected @param tag for 'p1''
```

please use "mX" for name of of method to keep them short vertically.
plesae use `// X violations above|below:`

**Review by romani — CHANGES_REQUESTED** [review]
it was requested before but somehow missed.

last:

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocmethod/Example1.java (hunk 7 above)
Comment (AmitKumarDeoghoria) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocmethod/Example1.java]: Done

**Comment by AmitKumarDeoghoria** [comment]
GitHub, generate website

**Review by romani — APPROVED** [review]
Thanks a lot!!!

**Pull request merged**